src/hipscat_import/margin_cache/margin_cache.py

```diff
@@ -145,5 +145,7 @@ def generate_margin_cache(args, client):
             catalog_base_dir=args.catalog_path, dataset_info=margin_catalog_info
         )
         step_progress.update(1)
+        cache_path = mcmr.get_cache_directory(args.catalog_path)
         file_io.remove_directory(args.tmp_path, ignore_errors=True)
+        file_io.remove_directory(cache_path, ignore_errors=True)
```

Comment:
Hm. It looks like the margin cache process doesn't make use of the `args.tmp_path` at all, and would have been cleaned up by the above step, if it were. I created a branch that uses the `tmp_path` instead (https://github.com/astronomy-commons/hipscat-import/tree/delucchi/margin_tmp). Can you see if that addresses the empty shard dir problem? I'd like users to be able to use a different path for tmp files (see this doc: https://hipscat-import.readthedocs.io/en/latest/catalogs/temp_files.html)

Reply:
Ah I see! That makes sense, and yeah I tested with that branch and it solves the issue. I can close this PR and we can go with that solution if you think that's better.

Reply:
Sounds good. Sending you the PR.